From 8c04fc5ade094c2150ecd25cf4842b335b3bbdcc Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Wed, 12 Feb 2020 14:29:34 +0100
Subject: [PATCH] PONRTSYS-6689: Remove LAN port HW specific ioctl interface

To avoid fuzz testing just remove the LAN port specific ioctl interface.
This is not used and removing this reduced the attack surface of the system.
---
 drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
index 67008ac67e9b..e61858c1720f 100644
--- a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
+++ b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
@@ -69,8 +69,6 @@ static int ltq_eth_stop(struct net_device *dev);
 static void ltq_eth_uninit(struct net_device *dev);
 /* Transmit packet from Tx Queue to MAC */
 static int ltq_start_xmit(struct sk_buff *skb, struct net_device *dev);
-/* Hardware specific IOCTL's  */
-static int ltq_ioctl(struct net_device *dev, struct ifreq *ifr, int cmd);
 /* Get the network device statistics */
 static
 struct rtnl_link_stats64 *ltq_get_stats(struct net_device *dev,
@@ -116,7 +114,6 @@ static const struct net_device_ops ltq_eth_drv_ops = {
 	.ndo_set_mac_address	= ltq_set_mac_address,
 	.ndo_change_mtu		= ltq_change_mtu,
 	.ndo_get_stats64	= ltq_get_stats,
-	.ndo_do_ioctl		= ltq_ioctl,
 	.ndo_tx_timeout		= ltq_tx_timeout,
 	.ndo_bridge_setlink	= switchdev_port_bridge_setlink,
 	.ndo_bridge_getlink	= switchdev_port_bridge_getlink,
@@ -1089,12 +1086,6 @@ static int ltq_start_xmit(struct sk_buff *skb, struct net_device *dev)
 	return 0;
 }
 
-/* Platform specific IOCTL's handler */
-static int ltq_ioctl(struct net_device *dev, struct ifreq *ifr, int cmd)
-{
-	return -EOPNOTSUPP;
-}
-
 /* init of the network device */
 static int ltq_eth_init(struct net_device *dev)
 {
